packages/framework/src/services/microservice/gateway/kafka-gateway.service.test.ts

```diff
@@ -0,0 +1,74 @@
+import * as sinon from 'sinon';
+import { suite } from 'uvu';
+import * as assert from 'uvu/assert';
+
+import { CompleteTransportQueueItem } from '../../../models';
+import { KafkaBrokerClient, KafkaConsumePayload } from '../../kafka/kafka-broker.client';
+import { KafkaGateway, KafkaGatewayService } from './kafka-gateway.service';
+
+const test = suite('Kafka Microservice Gateway');
+
+test.before.each((context) => {
+  context.sandbox = sinon.createSandbox();
+});
+
+test.after.each((context) => {
+  context.sandbox.restore();
+});
+
+test('should call the register correctly', async (context) => {
+  const kafkaClientConnectStub = context.sandbox.stub(KafkaBrokerClient.prototype, 'connect');
+
+  await KafkaGateway.register({ broker: 'localhost:9092', transport: 'kafka' });
+  assert.is(kafkaClientConnectStub.called, true);
+});
+
+test('should call the subscribe correctly', (context) => {
+  const cb: KafkaConsumePayload = { topic: 'text_event', message: new Buffer(`{ "data": "test" }`) };
```

Comment:
Let's use Buffer.from as mentioned above
```suggestion
  const cb: KafkaConsumePayload = {
    topic: 'text_event',
    message: Buffer.from(`{ "data": "test" }`, 'utf-8'),
  };
```